adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBBannerRenderer.m

```diff
@@ -0,0 +1,86 @@
+//
+//  GADPangleRTBBannerRenderer.m
+//  PangleAdapter
+//
+//  Created by bytedance on 2022/1/11.
+//
+
+#import "GADPangleRTBBannerRenderer.h"
+#import "GADMediationAdapterPangleConstants.h"
+#import <BUAdSDK/BUAdSDK.h>
+#import "GADMAdapterPangleUtils.h"
+
+@interface GADPangleRTBBannerRenderer() <BUNativeExpressBannerViewDelegate>
+
+@end
+
+@implementation GADPangleRTBBannerRenderer {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationBannerLoadCompletionHandler _loadCompletionHandler;
+    /// The Pangle banner ad.
+    BUNativeExpressBannerView *_nativeExpressBannerView;
+    /// An ad event delegate to invoke when ad rendering events occur.
+    id<GADMediationBannerAdEventDelegate> _delegate;
+    /// The requested ad size.
+    CGSize _bannerSize;
+}
+
+- (void)renderBannerForAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration
+                     completionHandler:
+(nonnull GADMediationBannerLoadCompletionHandler)completionHandler {
+    _loadCompletionHandler = completionHandler;
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterPanglePlacementID] ?: @"";
+    if (PangleIsEmptyString(placementId)) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorInvalidServerParameters, [NSString stringWithFormat:@"%@ cannot be nil.",GADMAdapterPanglePlacementID]);
+        _loadCompletionHandler(nil, error);
+        return;
+    }
+    _bannerSize = adConfiguration.adSize.size;
+    _nativeExpressBannerView = [[BUNativeExpressBannerView alloc]initWithSlotID:placementId rootViewController:adConfiguration.topViewController adSize:adConfiguration.adSize.size];
+    _nativeExpressBannerView.delegate = self;
+    if (![_nativeExpressBannerView respondsToSelector:@selector(setAdMarkup:)]) {
+        NSError *error = GADMAdapterPangleErrorWithCodeAndDescription(GADPangleErrorVersionLow, @"Pangle SDK version is too low,please update Pangle SDK to the latest version.");
+        _loadCompletionHandler(nil, error);
+        return;
```

Comment:
Question, does the publisher need to do import Pangle SDK separately? The adapter should already include Pangle SDK as the dependency.

Reply:
The publisher doesn't need to do import Pangle SDK. The adapter should include Pangle SDK version >= 4300.

Reply:
I saw the latest version of `Ads-Global` is 4.3.0.2, see podspec:
https://github.com/bytedance/Bytedance-UnionAD/blob/master/Ads-Global/Ads-Global.podspec

Which means this check can be removed?

Reply:
Gentle bump on this. Since the adapter will use `4.3.0.2`, this check can be removed.